pkg/utils/watchJob.go

```diff
@@ -83,3 +86,27 @@ func JobLogs(experimentName string, engineName string, client *kubernetes.Client
 	}
 	return 0, nil
 }
+
+//ResultUpdate will update the result of pipelines in a table by calling a python script result_update.py
+func ResultUpdate(experimentName string, engineName string, clientSet *chaosClient.LitmuschaosV1alpha1Client) (int, error) {
+
+	//Updating the result table
+	dt := time.Now()
+	app, err := clientSet.ChaosResults(chaosTypes.ChaosNamespace).Get(engineName+"-"+experimentName, metav1.GetOptions{})
+	Expect(err).To(BeNil(), "Fail to get the chaosresult while updating the result in a table")
```

Comment:
for pkg function use the normal error: i.e: like `if err != nil {`

Reply:
@chandankumar4  for using `if err != nil ` I have to add `return` and this return will give the different reason for failure.

Suppose At different places it is failing due to different reason then It will be difficult to get the exact reason. 

For Details:
Here is how I'm calling this function and :
```
//Updating the result table
By("Updating the result table")
pipelineResult, err := utils.ResultUpdate(experimentName, engineName, clientSet)
Expect(pipelineResult).NotTo(Equal("1"), "Failed  to update the job result in a table")
Expect(err).To(BeNil(), "Fail run the script for result updation")
fmt.Println("Result updated successfully !!!")
```
But the exact reason is it is not getting chaos result.

Reply:
you can use formated error return statement also

Reply:
okay I'm using  formated error now logs after this
```
    Should check for the result updation [It]
    /home/gitlab-runner/go/src/github.com/litmuschaos/litmus-e2e/tests/container-kill_test.go:174

    Fail run the script for result updation
    Expected
        <*errors.withStack | 0xc0005dc860>: {
            error: {
                cause: {
                    ErrStatus: {
                        TypeMeta: {Kind: "", APIVersion: ""},
                        ListMeta: {
                            SelfLink: "",
                            ResourceVersion: "",
                            Continue: "",
                            RemainingItemCount: nil,
                        },
                        Status: "Failure",
                        Message: "chaosresults.litmuschaos.io \"engine1-container-kill\" not found",
                        Reason: "NotFound",
                        Details: {
                            Name: "engine1-container-kill",
                            Group: "litmuschaos.io",
                            Kind: "chaosresults",
                            UID: "",
                            Causes: nil,
                            RetryAfterSeconds: 0,
                        },
                        Code: 404,
                    },
                },
                msg: "Fail to get the chaosresult while updating the result in a table.",
            },
            stack: [0x1091e1c, 0x10ff0a9, 0x10acc0c, 0x10ac87f, 0x10abd14, 0x10b2ce6, 0x10b2594, 0x10b852f, 0x10b8044, 0x10b7887, 0x10b9ece, 0x10bca37, 0x10bc6af, 0x10fbde4, 0x4f9440, 0x45e881],
        }
    to be nil
```

Reply:
In this file, there are more places where BDD type error are used you can change that later also